frontend/src/main/scala/akkaviz/frontend/ApiConnection.scala

```diff
@@ -11,30 +11,41 @@ import scala.scalajs.js.typedarray.TypedArrayBufferOps
 import scala.scalajs.js.{JavaScriptException, timers}
 
 trait Upstream {
-  def send(msg: String): Unit
+  def send(msg: ApiClientMessage): Unit
+
+  def onclose: js.UndefOr[js.Function1[CloseEvent, _]]
+
+  def onclose_=(fn: js.Function1[CloseEvent, _]): Unit
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method name does not match the regular expression '^([a-z][A-Za-z0-9]*|<<|>>>|>|==|!=|<|<=|>|>=|\||&|\^|\+|-|\*|\/|%|\|\||&&|\+\+|--|\+=|-=)$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2183085963/issues/source?bid=3121282&fileBranchId=3164965#l18)